Uni_life_tracker/app/src/main/java/com/generic/ult/Add_expenses.java

```diff
@@ -0,0 +1,56 @@
+package com.generic.ult;
+
+import android.os.Bundle;
+import android.view.View;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.navigation.ui.AppBarConfiguration;
+import com.google.android.material.textfield.TextInputLayout;
+
+import java.util.Objects;
+
+
+public class Add_expenses extends AppCompatActivity {
+
+    private AppBarConfiguration appBarConfiguration;
+
+    private TextInputLayout textInputName;
+    private TextInputLayout textInputAmount;
+    private TextInputLayout textInputDate;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_add_expenses);
+
+        textInputName = findViewById(R.id.add_e_name);
+        textInputAmount = findViewById(R.id.add_e_amount);
+        textInputDate = findViewById(R.id.add_e_date);
+
+
+    }
+
+
+    private boolean validate(TextInputLayout textInput) {
+        String Input = Objects.requireNonNull(textInput.getEditText()).getText().toString().trim();
+
+        if (Input.isEmpty()) {
+            textInput.setError("Field cannot be empty");
+            return false;
+        } else {
```

Comment:
I think it would be helpful to add en elseif for the date to check if the format is correct (similar to what I did with the password) I think there could be regexs that you could compare them to so it should be too difficult

Reply:
Or I think it might be easier to input a datepicker, so it would just show a calender instead